### PR TITLE
Cut v1.8.0

### PR DESCRIPTION
**Pull request by beorn7**
In particular, we want to pull in https://github.com/prometheus/common/pull/616

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/prometheus/pushgateway
 
-go 1.20
+go 1.21
+
+toolchain go1.22.0
```
Comment (SuperQ) [inline go.mod:5]: I've been skipping the new `toolchain` directive in our configs for now.

```suggestion
```

**Review by SuperQ — APPROVED** [review]
Minor nit about `toolchain`, otherwise LGTM.

**Comment by SuperQ** [comment]
Thanks for updating to the new version collector.

**Comment by beorn7** [comment]
I'll also rebase on top of the Go update. Thanks for that, too.

**Pull request merged**